src/components.ts

```diff
@@ -15,33 +15,43 @@ export enum MessageComponentTypes {
 
 export type MessageComponent = Button | ActionRow | StringSelect | InputText;
 
-/**
- * Button component
- * @see {@link https://discord.com/developers/docs/interactions/message-components#button-object-button-structure}
- */
-export type Button = {
-  type: MessageComponentTypes.BUTTON;
-  style:
-    | ButtonStyleTypes.PRIMARY
-    | ButtonStyleTypes.SECONDARY
-    | ButtonStyleTypes.SUCCESS
-    | ButtonStyleTypes.DANGER
-    | ButtonStyleTypes.LINK;
-  label: string;
-  emoji?: Pick<EmojiInfo, 'id' | 'name' | 'animated'>;
-  custom_id?: string;
-  url?: string;
-  disabled?: boolean;
-};
-
 export enum ButtonStyleTypes {
   PRIMARY = 1,
   SECONDARY = 2,
   SUCCESS = 3,
   DANGER = 4,
   LINK = 5,
+  PREMIUM = 6,
+}
+
+interface BaseButton {
+  disabled?: boolean;
+  emoji?: Pick<EmojiInfo, 'id' | 'name' | 'animated'>;
+  label: string;
+  type: MessageComponentTypes.BUTTON;
+}
+
+interface PremiumButton extends BaseButton {
+  sku_id: string;
+  style: ButtonStyleTypes.PREMIUM;
+}
+
+interface LinkButton extends BaseButton {
+  url: string;
+  style: ButtonStyleTypes.LINK;
+}
+
+interface AllOtherButtons extends BaseButton {
```

Comment:
Heh, this interface name doesn't feel quite right.  How about just `Button`?  Or if we really need to differentiate, `StandardButton`?

Reply:
hah yeah I don't like the name either. Button would be great but we export `Button` below. I'm fine with `StandardButton`

Reply:
how about `CustomButton`?

Reply:
That works for me!